--- pyproject.toml	2021-02-02 01:15:28.199269629 +0800
+++ pyproject.toml2	2021-02-02 01:14:03.655751029 +0800
@@ -1,3 +1,15 @@
+[build-system]
+requires = [ 
+  "setuptools>=51", 
+  "wheel>=0.36",
+  "setuptools_scm[toml]>=5.0",
+  "pybind11_generics @ git+https://github.com/bluecheetah/pybind11_generics.git@0.3.0"
+  ]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
+write_to = "src/pybind11_generics_tests/_version.py"
+
 [tool.black]
 line-length = 100
 target-version = [ "py38",]
